qutebrowser/completion/completiondelegate.py

```diff
@@ -194,21 +208,15 @@ def _get_textdoc(self, index):
         self._doc.setDefaultTextOption(text_option)
         self._doc.setDocumentMargin(2)
 
-        assert _cached_stylesheet is not None
-        self._doc.setDefaultStyleSheet(_cached_stylesheet)
-
         if index.parent().isValid():
             view = self.parent()
             pattern = view.pattern
             columns_to_filter = index.model().columns_to_filter(index)
+            self._doc.setPlainText(self._opt.text)
             if index.column() in columns_to_filter and pattern:
-                repl = r'<span class="highlight">\g<0></span>'
-                pat = html.escape(re.escape(pattern)).replace(r'\ ', r'|')
-                txt = html.escape(self._opt.text)
-                text = re.sub(pat, repl, txt, flags=re.IGNORECASE)
-                self._doc.setHtml(text)
-            else:
-                self._doc.setPlainText(self._opt.text)
+                pat = re.escape(pattern).replace(r'\ ', r'|')
+                _Highlighter(self._doc, pat,
+                             config.val.colors.completion.match.fg)
```

Comment:
The type of the setting is currently a `QssColor` which is a string to be used in a stylesheet, but it should now be a `QtColor` instead (which gives you a `QColor` object).

See this failing test: https://travis-ci.org/qutebrowser/qutebrowser/jobs/431903868#L721

Reply:
```
class QtColor(BaseType):

    """A color value.

    A value can be in one of the following formats:

    * `#RGB`/`#RRGGBB`/`#RRRGGGBBB`/`#RRRRGGGGBBBB`
    * An SVG color name as specified in
      http://www.w3.org/TR/SVG/types.html#ColorKeywords[the W3C specification].
    * transparent (no color)
    """

class QssColor(BaseType):

    """A color value supporting gradients.

    A value can be in one of the following formats:

    * `#RGB`/`#RRGGBB`/`#RRRGGGBBB`/`#RRRRGGGGBBBB`
    * An SVG color name as specified in
      http://www.w3.org/TR/SVG/types.html#ColorKeywords[the W3C specification].
    * transparent (no color)
    * `rgb(r, g, b)` / `rgba(r, g, b, a)` (values 0-255 or percentages)
    * `hsv(h, s, v)` / `hsva(h, s, v, a)` (values 0-255, hue 0-359)
    * A gradient as explained in
      http://doc.qt.io/qt-5/stylesheet-reference.html#list-of-property-types[the Qt documentation] under ``Gradient''
    """
```

Looks like `QssColor` accepts more formats. Do I need to parse it into a `QtColor` to keep backwards compatibility? Or can I just change the setting type and accept that configs using `hsv` or `rgb` for this setting might break?

Reply:
Hmm, good point... Gradient's won't work anymore either way, but that's fine.

I'd kinda like to have support for those functions in `QtColor`, also for the additional consistency between `QtColor` and `QssColor`. I suppose it shouldn't be too hard to parse using some regex.

FWIW here's Qt's parsing for it, but it uses their CSS(-ish) parser which isn't exposed: https://github.com/qt/qtbase/blob/v5.11.2/src/gui/text/qcssparser.cpp#L711-L778

Do you want to take a look at that, @rcorre? If you don't have the time I can do so, but I have lots of PRs to look at still :wink:

Reply:
Gotcha, I'll take a look.